### ISSUE TITLE
[BUG]: Custom bond features for 1-atom, 0-bond molecule

### ISSUE THREAD

**Issue by hesther** [chemprop/chemprop#261]
**Describe the bug**
Custom bond features do not work if the dataset contains a 1-atom molecule (that has zero bonds).

**Example(s)**
Use the dummy dataset 
```
smiles, target
C, 1
N, 0.01
CC, 2
```
and save as test.csv, then run 

```
import pandas as pd
from rdkit import Chem
import numpy as np
from chemprop.features import bond_features

smiles = pd.read_csv("test.csv")['smiles'].values

descriptors=[]
for smi in smiles:
    mol = Chem.MolFromSmiles(smi)
    features=[]
    for bond in mol.GetBonds():
        features.append(bond_features(bond))
    features=np.array(features)
    descriptors.append(features)
    
np.savez('descriptors_bond2.npz', *descriptors)
```
to make bond features with bond_features() from chemprop.featurization and saved them to npz (I wanted to attach the npz file but GitHub won't let me). Then run

`chemprop_train --data_path test.csv --dataset_type regression`

works as expected, but 

`chemprop_train --data_path test.csv --dataset_type regression --bond_features_path descriptors_bond2.npz --overwrite_default_bond_features --no_bond_features_scaling`

fails. It works if test.csv does not contain 1-atom molecules (feel free to try). Technically, this should be the exact same as the first command since we just read in the bond features manually (but they are exactly the same)

**Expected behavior**
Should yield approximately the same model no matter whether the same bond features are created by Chemprop internally, or manually and loaded into descriptors_bond2.npz

**Checklist**
- [x] all dependencies are satisifed: `conda list` shows the packages listed in the `README`
- [x] the unit tests are working: `pytest -v` reports no errors

**Comment by davidegraff** [comment]
ideally need a stack trace to decipher where this is going wrong to properly diagnose it

**Comment by kevingreenman** [comment]
We are closing all bug reports that we don't have time to fix before the release of v1.7 this month, and unfortunately this is one of them. We will soon release [Chemprop v2.0](https://github.com/chemprop/chemprop/issues/517) and do not plan to maintain/support v1.x any longer. If anyone has time to open a PR that fixes this bug before the end of January 2024, we will review it and consider including it in v1.7.
